File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/data/viewmodels/PermissionsViewModel.java
package com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels;

import androidx.lifecycle.ViewModel;

public class PermissionsViewModel extends ViewModel {

    private boolean recordAudioAllowed = false;

    public boolean isRecordAudioAllowed() {
        return recordAudioAllowed;
    }

    public void setRecordAudioAllowed(boolean recordAudioAllowed) {
        this.recordAudioAllowed = recordAudioAllowed;
    }
}

File: app/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/data/viewmodel/datastore/dsvolatile/PermissionsViewModel.kt
package com.tritiumgaming.phasmophobiaevidencepicker.data.viewmodel.datastore.dsvolatile

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class PermissionsViewModel: ViewModel() {

    var isRecordAudioAllowed: Boolean = false

    class PermissionsFactory(

    ) : ViewModelProvider.Factory {

        override fun <T : ViewModel> create(modelClass: Class<T>): T {
            if (modelClass.isAssignableFrom(PermissionsViewModel::class.java)) {
                @Suppress("UNCHECKED_CAST")
                return PermissionsViewModel(

                ) as T
            }
            throw IllegalArgumentException("Unknown ViewModel class")
        }
    }

}
